tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,2041 @@
+            return $"{TestKeyPrefix}-{testName}-{Guid.NewGuid().ToString("N").Substring(0, 8)}";
+        }
+
+        private async Task<string> CreateSnapshot(string snapshotName, IEnumerable<ConfigurationSettingsFilter> settingsToInclude, CancellationToken cancellationToken = default)
+        {
+            ConfigurationSnapshot snapshot = new ConfigurationSnapshot(settingsToInclude);
+
+            snapshot.SnapshotComposition = SnapshotComposition.Key;
+
+            CreateSnapshotOperation operation = await _configClient.CreateSnapshotAsync(
+                WaitUntil.Completed,
+                snapshotName,
+                snapshot,
+                cancellationToken);
+
+            return operation.Value.Name;
+        }
+
+        public async Task InitializeAsync()
+        {
+            DefaultAzureCredential credential = _defaultAzureCredential;
+            string subscriptionId = GetCurrentSubscriptionId();
+
+            var armClient = new ArmClient(credential);
+            SubscriptionResource subscription = armClient.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{subscriptionId}"));
+
+            ResourceGroupResource resourceGroup = await subscription.GetResourceGroups().GetAsync(ResourceGroupName);
+
+            AppConfigurationStoreResource appConfigStore = null;
+
+            try
+            {
+                appConfigStore = await resourceGroup.GetAppConfigurationStores().GetAsync(AppConfigStoreName);
+            }
+            catch (RequestFailedException ex) when (ex.Status == 404)
+            {
+                throw new InvalidOperationException($"App Configuration store '{AppConfigStoreName}' not found in resource group '{ResourceGroupName}'. Please create it before running tests.", ex);
+            }
+
+            AsyncPageable<AppConfigurationStoreApiKey> accessKeys = appConfigStore.GetKeysAsync();
+
+            _connectionString = (await accessKeys.FirstAsync()).ConnectionString;
+
+            _configClient = new ConfigurationClient(_connectionString);
+
+            KeyVaultResource keyVault = null;
+
+            // Find and initialize Key Vault - look in the same resource group
+            keyVault = await resourceGroup.GetKeyVaults().GetAsync(KeyVaultName);
+
+            if (keyVault == null)
+            {
+                throw new InvalidOperationException(
+                    $"Key Vault '{KeyVaultName}' not found in subscription {subscriptionId}. " +
+                    "This resource is required for integration tests. " +
+                    "Please create the Key Vault with the appropriate permissions before running tests.");
+            }
+
+            _keyVaultEndpoint = keyVault.Data.Properties.VaultUri;
+
+            _secretClient = new SecretClient(_keyVaultEndpoint, credential);
+
+            Console.WriteLine($"Successfully connected to App Configuration store '{AppConfigStoreName}' and Key Vault '{KeyVaultName}'");
```

Comment:
XUnit exposes some type like ITestOutputHelper that's supposed to replace Console when writing messages during tests.